File: apps/apple/TopsyTurvy/Theatre/Engine/AnalysisResult.swift
/// Represents an analysis result returned as JSON by `topsyturvy_analyse` from the Topsy Turvy toolchain.
struct AnalysisResult: Decodable {
    /// A value indicating whether the analysis was successful.
    let Success: Bool

    /// The diagnostics reported by the analysis.
    let Diagnostics: [DiagnosticInfo]
}

File: operetta/BWHazel.TopsyTurvy.Embedded/Analysis/AnalysisResult.cs
using System.Collections.Generic;

namespace BWHazel.TopsyTurvy.Embedded.Analysis;

/// <summary>
/// Represents the result of a parse and type-check pass, returned as JSON by <see cref="NativeExports.ToolchainExports.AnalyseSource"/>.
/// </summary>
/// <param name="Success">A value indicating whether parsing and type-checking both succeeded with no error-severity diagnostics.</param>
/// <param name="Diagnostics">The diagnostics produced by parsing and type-checking, in the order produced.</param>
public record AnalysisResult(bool Success, IReadOnlyList<DiagnosticInfo> Diagnostics);
